backend/capellacollab/sessions/routes.py

```diff
@@ -87,70 +100,72 @@ def get_current_sessions(
 
 
 @router.post(
-    "/",
+    "/readonly",
     response_model=AdvancedSessionResponse,
 )
 def request_session(
-    body: PostSessionRequest,
+    body: PostReadonlySessionRequest,
     db_user: DatabaseUser = Depends(get_own_user),
     operator: Operator = Depends(get_operator),
     db: Session = Depends(get_db),
     token=Depends(JWTBearer()),
 ):
-    assert body.type == WorkspaceType.READONLY
-
     rdp_password = generate_password(length=64)
 
-    owner = db_user.username
+    owner = db_user.name
 
     log.info("Starting persistent session creation for user %s", owner)
 
-    existing_user_sessions = database.get_sessions_for_user(db, owner)
+    project = get_project_by_slug(db, body.project_slug)
+    if not project:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            {
+                "reason": f"The project with name {body.project_slug} was not found.",
+                "technical": f"No project {body.project_slug} found.",
+            },
+        )
 
-    if body.repository in [
-        session.repository for session in existing_user_sessions
-    ]:
+    verify_project_role(project.name, token, db)
+
+    model = get_model_by_slug(db, body.project_slug, body.model_slug)
+    if not model:
         raise HTTPException(
-            status_code=404,
-            detail={
-                "err_code": "existing_session",
-                "reason": f"You already have a open Read-Only Session for the repository {body.repository}. Please navigate to 'Active Sessions' to Reconnect",
+            status.HTTP_404_NOT_FOUND,
+            {
+                "reason": f"The model with name {body.model_slug} was not found.",
+                "technical": f"No model {body.model_slug} found.",
             },
         )
-    verify_project_role(repository=body.repository, token=token, db=db)
-    git_model = git_models_crud.get_primary_model_of_repository(
-        db, body.repository
-    )
-    if not git_model:
+
+    models = [
+        m
+        for m in project.models
+        if m.git_models and m.version.id == model.version.id
+    ]
+    if not models:
         raise HTTPException(
-            status_code=404,
+            status_code=status.HTTP_404_NOT_FOUND,
             detail={
                 "err_code": "git_model_not_found",
-                "reason": "The Model has no connected Git Model. Please contact a project manager or admininistrator",
+                "reason": "The Project has no connected Git Model. Please contact a project manager or admininistrator",
             },
         )
 
-    revision = body.branch or git_model.revision
-    if body.depth == DepthType.LatestCommit:
-        depth = 1
-    elif body.depth == DepthType.CompleteHistory:
-        depth = 0
-    else:
+    docker_image = get_readonly_image_for_version(model.version)
```

Comment:
I'm not able to set the `readonly` image. To reproduce: 
Navigate to http://localhost:4200/settings/core/tool/1, change the read-only image and refresh the page. It shows the default again. 
I couldn't implement it due to the missing template in the database, but it should be an easy fix. Check the `FIXME` in capellacollab/tools/crud.py

Reply:
I tried and it seems fixed now.